File: fix_tabs.js
const fs = require('fs');
let html = fs.readFileSync('private/dashboard.html', 'utf8');

// Use regex to remove all tab-btn inline styles
html = html.replace(/\s*#scoreboard-section \.tab-btn \{[\s\S]*?\}/g, '');
html = html.replace(/\s*#scoreboard-section \.tab-btn:hover \{[\s\S]*?\}/g, '');
html = html.replace(/\s*#scoreboard-section \.tab-btn\.active \{[\s\S]*?\}/g, '');

fs.writeFileSync('private/dashboard.html', html);
console.log('Tabs fixed');

File: fix_tabs.py
import re

with open('/home/harry/Projects/ctf/private/dashboard.html', 'r') as f:
    content = f.read()

# Inject showSection function at the very beginning of the MAIN SCRIPT
inject = """  <script>
    function showSection(sectionId) {
      document.querySelectorAll('.info-section').forEach(sec => sec.style.display = 'none');
      const target = document.getElementById(sectionId);
      if(target) target.style.display = 'block';
      
      document.querySelectorAll('.nav-link').forEach(link => link.classList.remove('active'));
      const activeLinks = document.querySelectorAll(`.nav-link[onclick*="${sectionId}"]`);
      activeLinks.forEach(l => l.classList.add('active'));
    }
"""

content = content.replace("  <script>", inject, 1)

with open('/home/harry/Projects/ctf/private/dashboard.html', 'w') as f:
    f.write(content)
